### PR TITLE
Replace `log` package instances with appropriate mlog `LogX` method.

### PR DESCRIPTION
**Pull request by haardikdharma10**
#### Summary

Scanned `plugin/client_rpc.go` and replaced instances that use std `log` package with mlog `LogX` method.

#### Ticket Link

  Fixes https://github.com/mattermost/mattermost-server/issues/17650
  JIRA https://mattermost.atlassian.net/browse/MM-35890

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by haardikdharma10** [comment]
@hanzei, can you check in here on why the tests fail ? 🤔

File: plugin/client_rpc.go [hunk 1]
```diff
@@ -288,14 +288,14 @@ func (g *hooksRPCClient) ServeHTTP(c *Context, w http.ResponseWriter, r *http.Re
 	go func() {
 		connection, err := g.muxBroker.Accept(serveHTTPStreamId)
 		if err != nil {
-			g.log.Error("Plugin failed to ServeHTTP, muxBroker couldn't accept connection", mlog.Uint32("serve_http_stream_id", serveHTTPStreamId), mlog.Err(err))
+			mlog.Error("Plugin failed to ServeHTTP, muxBroker couldn't accept connection", mlog.Uint32("serve_http_stream_id", serveHTTPStreamId), mlog.Err(err))
```
[the remark this answers is not in the document]
Comment (hanzei) [reply plugin/client_rpc.go:291]: The same applies to all usages of `mlog`

[the remark this answers is not in the document]
Comment (haardikdharma10) [reply plugin/client_rpc.go]: Thanks:) Applied the necessary changes!

File: plugin/client_rpc.go [hunk 2]
```diff
@@ -242,10 +242,10 @@ func (g *apiRPCClient) LoadPluginConfiguration(dest interface{}) error {
 	_args := &Z_LoadPluginConfigurationArgsArgs{}
 	_returns := &Z_LoadPluginConfigurationArgsReturns{}
 	if err := g.client.Call("Plugin.LoadPluginConfiguration", _args, _returns); err != nil {
-		log.Printf("RPC call to LoadPluginConfiguration API failed: %s", err.Error())
+		g.log.Error("RPC call to LoadPluginConfiguration API failed: ", mlog.Err(err))
```
Comment (hanzei) [inline plugin/client_rpc.go:245]: The `apiRPCClient` only has a `g.LogError` method.

`g.log` is only available for `hooksRPCClient`

**Review by hanzei — CHANGES_REQUESTED** [review]
There are some `fmt.Fprintf(os.Stderr` call leaved. Can these also get migrated?

**Comment by haardikdharma10** [comment]
> There are some `fmt.Fprintf(os.Stderr` call leaved. Can these also get migrated?

Sure @hanzei, can you help me with an example on how they should be migrated?

File: plugin/client_rpc.go (hunk 2 above)
Comment (haardikdharma10) [reply plugin/client_rpc.go:245]: Done!

**Comment by haardikdharma10** [comment]
@hanzei, all checks are passing :) 🎉

File: plugin/client_rpc.go [hunk 3]
```diff
@@ -62,8 +62,9 @@ type apiRPCClient struct {
 }
 
 type apiRPCServer struct {
-	impl      API
-	muxBroker *plugin.MuxBroker
+	impl         API
+	muxBroker    *plugin.MuxBroker
+	apiRPCClient *apiRPCClient
```
Comment (hanzei) [inline plugin/client_rpc.go:67]: Is this every assigned somewhere?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: plugin/client_rpc.go (hunk 3 above)
Comment (haardikdharma10) [reply plugin/client_rpc.go:67]: yes, on line 771!

Comment (haardikdharma10) [reply plugin/client_rpc.go:67]: Hey @hanzei, wondering if you have got time to come back to this ? 😄

Comment (hanzei) [reply plugin/client_rpc.go:67]: In L 771 you access `apiRPCClient`. I was wondering where you _assign_ a value to it.

Comment (haardikdharma10) [reply plugin/client_rpc.go:67]: Whoops, yeah. so I should assign it something like this :

```
s.apiRPCClient = &apiRPCClient{
		client:    rpc.NewClient(connection),
		muxBroker: s.muxBroker,
	}
```
in the function `InstallPlugin`? Does that sound right to you @hanzei ?

Comment (hanzei) [reply plugin/client_rpc.go:67]: After re-reading the code, I don't think that calling a log method on the client side is a good idea. if `s.muxBroker.Dial` the RPC connection won't work and hence calling the client side logging method won't neither.

Either we should stick with `fmt.Fprintf` or just `mlog`

File: plugin/client_rpc.go [hunk 4]
```diff
@@ -396,7 +397,7 @@ func (g *apiRPCClient) PluginHTTP(request *http.Request) *http.Response {
 
 	requestBody, err := ioutil.ReadAll(request.Body)
 	if err != nil {
-		log.Printf("RPC call to PluginHTTP API failed: %s", err.Error())
+		g.LogError("RPC call to PluginHTTP API failed: ", err.Error())
```
Comment (hanzei) [inline plugin/client_rpc.go:400]: `LogError` logs first a message and then a list of pairs. Hence this needs to be something like:
```suggestion
		g.LogError("RPC call to PluginHTTP API failed", "error", err.Error())
```

The same comment applies to some other places. Please double check the file.

**Review by hanzei — CHANGES_REQUESTED** [review]

File: plugin/client_rpc.go (hunk 3 above)
Comment (haardikdharma10) [reply plugin/client_rpc.go]: Thanks. For now, i have kept it to `fmt.Fprintf`

File: plugin/client_rpc.go (hunk 4 above)
Comment (haardikdharma10) [reply plugin/client_rpc.go]: Done! Thanks @hanzei!

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Review by larkox — DISMISSED** [review]

**Comment by larkox** [comment]
@haardikdharma10 Not sure who would be the QA person for this ticket, so I leave it to you to assign it.

**Comment by haardikdharma10** [comment]
Thanks @larkox, even I am not very sure. @hanzei, can you confirm if this needs a QA review, if yes, who can be the right person for a review?

**Comment by hanzei** [comment]
Great question @haardikdharma10. This PR should be double checked by QA to ensure no regression are introduced. @DHaussermann Could you please test this PR alongside with some plugin how fail to start for various reasons e.g. missing configuration and who panic at some point? You will need to replace the go dependency for `mattermost-server` and re-compile against this branch. Let me know if you have any questions.

**Comment by haardikdharma10** [comment]
A conflict was waiting to be resolved here. Done now! cc @DHaussermann

**Comment by isacikgoz** [comment]
Hi @haardikdharma10, big thanks for your contribution! There hasn't been recent activity on this PR so I've marked it as inactive. Following our [inactive contribution process](https://developers.mattermost.com/contribute/getting-started/inactive-contributions/), the PR is now closed due to 30 days or more of inactivity.

However, if you're still interested working on the changes, let us know! Note that we [merged multiple repositories into one monorepo](https://github.com/mattermost/mattermost-server/issues/22420), which may require you to resubmit the changes. 

If you have any questions, don't hesitate to let us know, we're happy to help.

**Pull request closed without merging**